shared_model/utils/variant_deserializer.hpp

```diff
@@ -18,8 +18,16 @@
 #ifndef IROHA_VARIANT_DESERIALIZER_HPP
 #define IROHA_VARIANT_DESERIALIZER_HPP
 
+#include <stdnoreturn.h>
 #include <boost/serialization/variant.hpp>
 
+// http://en.cppreference.com/w/c/language/_Noreturn
+#ifdef noreturn
+#define NORETURN noreturn
+#else
+#define NORETURN [[noreturn]]
```

Comment:
[It safe](http://en.cppreference.com/w/cpp/language/attributes) to use only that